lib/grape/named_route_matcher.rb

```diff
@@ -0,0 +1,25 @@
+module Grape
+  module NamedRouteMatcher
+    def method_missing(method_id, *arguments)
+      segments = arguments.first || {}
+
+      route = Grape::API.all_routes.detect do |r|
+        route_match?(r, method_id, segments)
```

Comment:
Could this `method_missing` use the "helper name" matching before do its job? It's too greedy for me. Seems you already have this method https://github.com/intridea/grape/pull/983/files#diff-64d352c6d46153e1e4ff964b0ff4ecdeR46 , maybe this matching can be done using this info?